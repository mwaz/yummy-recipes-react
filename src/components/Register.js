import React, { Component } from 'react';
import { Panel, Col, FormGroup, FormControl, Button, InputGroup } from 'react-bootstrap';
import "../index.css";
import background from "../images/recipes.jpg";

const registrationForm = {
    width: '100%',
    height: "100vh",
    paddingRight: "12%",
    paddingLeft: "52%",
    paddingTop: "100px",
    paddingBottom: "50px",
    backgroundColor: "#000",
    zIndex: "9999",
    filter: "opacity(.92)",
};
const registration = {
    width: '100%',
    height: "vh",
    backgroundColor: "#ffffff",
    filter: "opacity(.93)",
    
};

const backgroundDiv = {
    width: "100%",
    color: "white",
    height: "100vh",
    backgroundImage: "url(" + background + ")",
    backgroundRepeat: "repeat -y",
    zIndex: "-9999",

};

class Register extends Component {
    render() {
        return (
        <div style={backgroundDiv}>
            <div style={registration}>
            <div style={registrationForm}>
            <center><strong> <div style={{fontSize:"20px", paddingBottom:"1%"}}> Register Here </div></strong></center>
            <Panel header='Register' bsStyle="warning">
                <FormGroup>
                    <FormControl style={{backgroundColor:"black", color:"white", filter: "opacity(1)",}} type="text" id="username" placeholder="username" onChange={(event) => this.setState({ username: event.target.value })} />
                </FormGroup>
                <FormGroup>
                    <FormControl style={{backgroundColor:"black", color:"white", filter: "opacity(1)",}} type="email" id="email" placeholder="email" onChange={(event) => this.setState({ email: event.target.value })} />
                </FormGroup>
                <FormGroup>
                    <FormControl style={{backgroundColor:"black", color:"white", filter: "opacity(1)",}} type="password" id="password" placeholder="Password" onChange={(event) => this.setState({ password: event.target.value })} />
                </FormGroup>
                <FormGroup>
                    <FormControl style={{backgroundColor:"black", color:"white", filter: "opacity(1)",}} type="password" id="cpassword" placeholder="confirm Password" onChange={(event) => this.setState({ cpassword: event.target.value })} />
                </FormGroup>
                <Button bsStyle="success" onClick={(event => this.handleClick(event))}>Register</Button>
            </Panel>
            </div>
            </div>
            </div>
            

        
        );
    }
}
export default Register;
