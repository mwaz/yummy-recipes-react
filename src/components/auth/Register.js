import React, { Component } from 'react';
import { Panel, FormGroup, FormControl, Button } from 'react-bootstrap';
import { Redirect } from 'react-router-dom';
import { ToastContainer, toast } from 'react-toastify';
import NavigationBar from '../common/NavigationBar';
import Footer from '../common/Footer';
import axiosInstance from '../common/AxiosInstance';

/**
 * Component that handles user registration
 */
class Register extends Component {
  constructor(props) {
    super(props);
    /**
     * Sets different states for variables to be used in the component
     */
    this.state = {
      username: '',
      email: '',
      password: '',
      cpassword: '',
      redirect: '',
      errors: {},
    };
  }
  /**
     * Handles the click event genereated by the registration button
     */
  handleClick() {
    const payload = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      cpassword: this.state.cpassword,

    };
    if (this.state.password !== this.state.cpassword) {
      toast('Password Mismatch', { type: toast.TYPE.ERROR });
      return 0;
    }

    /**
     * Calls an axios instance to make the call to the API
     */
    axiosInstance
      .post('auth/register', payload)
      .then((response) => {
        toast.success(response.data.message);
        this.setState({ redirect: true });
        window.localStorage.setItem('login', 'False');

        console.log(JSON.stringify(response));
      })
      .catch((error) => {
        if (error.response) {
          toast.error(error.response.data['message']);
        }
      });
    return 0;
  }

  render() {
    /**
     * Renders the login page on successful registration
     * Can only render th registration page if and only if the token
     * does not exist in the local storage
     */
    const redirect = this.state.redirect;
    const token = window.localStorage.getItem('token');
    if (redirect || token) {
      return <Redirect to={{ pathname: '/categories' }} />;
    }
    return (
      <div>
        <div className="background-div">
          <NavigationBar />
          <div className="background-container">
            <div className="background-container-form">
              <center><strong> <div style={{ fontSize: '20px', paddingBottom: '1%' }}> Register Here </div></strong></center>
              <ToastContainer />
              <div className="card">
                <div className="card-title">Registration </div>
                <div className="card-text">
                  <Panel header="Register" bsStyle="warning">
                    <FormGroup>
                      <FormControl style={{ backgroundColor: 'white', color: 'black', filter: 'opacity(1)' }} type="text" id="username" placeholder="username" name="username" onChange={event => this.setState({ username: event.target.value })} />
                    </FormGroup>
                    <FormGroup>
                      <FormControl style={{ backgroundColor: 'white', color: 'black', filter: 'opacity(1)' }} type="email" id="email" placeholder="email" name="email" onChange={event => this.setState({ email: event.target.value })} />
                    </FormGroup>
                    <FormGroup>
                      <FormControl style={{ backgroundColor: 'white', color: 'black', filter: 'opacity(1)' }} type="password" id="password" placeholder="Password" name="password" onChange={event => this.setState({ password: event.target.value })} />
                    </FormGroup>
                    <FormGroup>
                      <FormControl style={{ backgroundColor: 'white', color: 'black', filter: 'opacity(1)' }} type="password" id="cpassword" placeholder="confirm Password" name="cpassword" onChange={event => this.setState({ cpassword: event.target.value })} />
                    </FormGroup>
                    <Button type="submit" bsStyle="success" onClick={(event => this.handleClick(event))}>Register</Button>
                  </Panel>

                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
export default Register;
