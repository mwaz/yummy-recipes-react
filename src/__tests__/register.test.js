import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Register from '../components/auth-front/register';
import toJson, { shallowToJson } from 'enzyme-to-json';

describe('Register component', () => {
    const register = jest.fn();
    const wrapper = shallow(<Register register={register} location={{}} />);
    const preventDefault = jest.fn();
    
    it('renders properly without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders all the divs', () =>{
        expect(wrapper.find('div').length).toBe(8)
    })

    it('renders the Form Group elements', () => {
        expect(wrapper.find('FormGroup').length).toBe(4)
        expect(wrapper.find('Button').simulate('submit', { preventDefault }))
    })

    it('has the correct form fields', () => {
        expect(wrapper.find('#username')).toHaveLength(1);
        expect(wrapper.find('#email')).toHaveLength(1);
        expect(wrapper.find('#password')).toHaveLength(1);
        expect(wrapper.find('#cpassword')).toHaveLength(1);
    });

    it('it renders state initially', () => {
        expect(wrapper.state().email).toEqual('');
        expect(wrapper.state().password).toEqual('');
        expect(wrapper.state().cpassword).toEqual('');
        expect(wrapper.state().username).toEqual('');
    });

    it('should create a user', () => {
        const component = shallow(<Register />);
        const preventDefault = jest.fn();
        component.setState({
           email: 'hey@hey.com',
           username: 'mwaz',
           password: '1234567',
           cpassword:'123456',
        });
        component.find('Button').simulate('submit', { preventDefault });
        expect(toJson(component)).toMatchSnapshot();
    });
    
    it('state changes on input change', () => {
        const component = shallow(<Register />);
        component.find('Button').simulate('change', {
            target: {
                value: 'Change function'
            }
        });

        expect(toJson(component)).toMatchSnapshot();
    });

});