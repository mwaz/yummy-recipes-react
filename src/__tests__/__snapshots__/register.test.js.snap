// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders properly without crashing 1`] = `
<div>
  <div
    className="background-div"
  >
    <NavigationBar />
    <div
      className="background-container"
    >
      <div
        className="background-container-form"
      >
        <center>
          <strong>
             
            <div
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "1%",
                }
              }
            >
               Register Here 
            </div>
          </strong>
        </center>
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <div
          className="card"
        >
          <div
            className="card-title"
          >
            Registration 
          </div>
          <div
            className="card-text"
          >
            <Uncontrolled(Panel)
              bsStyle="warning"
              header="Register"
            >
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="text"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="email"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="cpassword"
                  name="cpassword"
                  onChange={[Function]}
                  placeholder="confirm Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Register
              </Button>
            </Uncontrolled(Panel)>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Register component should create a user 1`] = `
<div>
  <div
    className="background-div"
  >
    <NavigationBar />
    <div
      className="background-container"
    >
      <div
        className="background-container-form"
      >
        <center>
          <strong>
             
            <div
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "1%",
                }
              }
            >
               Register Here 
            </div>
          </strong>
        </center>
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <div
          className="card"
        >
          <div
            className="card-title"
          >
            Registration 
          </div>
          <div
            className="card-text"
          >
            <Uncontrolled(Panel)
              bsStyle="warning"
              header="Register"
            >
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="text"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="email"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="cpassword"
                  name="cpassword"
                  onChange={[Function]}
                  placeholder="confirm Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Register
              </Button>
            </Uncontrolled(Panel)>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Register component state changes on input change 1`] = `
<div>
  <div
    className="background-div"
  >
    <NavigationBar />
    <div
      className="background-container"
    >
      <div
        className="background-container-form"
      >
        <center>
          <strong>
             
            <div
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "1%",
                }
              }
            >
               Register Here 
            </div>
          </strong>
        </center>
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <div
          className="card"
        >
          <div
            className="card-title"
          >
            Registration 
          </div>
          <div
            className="card-text"
          >
            <Uncontrolled(Panel)
              bsStyle="warning"
              header="Register"
            >
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="username"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="text"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="email"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="cpassword"
                  name="cpassword"
                  onChange={[Function]}
                  placeholder="confirm Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Register
              </Button>
            </Uncontrolled(Panel)>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
